### PR TITLE
Set up associations

### PR DESCRIPTION
**Pull request by ab-noori**
### 🌟Features:

> - [ ] - Add Book class with title and author properties, along with getters and setters for these attributes.
> - [ ] - Introduce the Rental class with date property and corresponding getter and setter methods.
> - [ ] - Create Classroom class with label property, has-many relationships with Students, and bidirectional association between Student and Classroom.
> - [ ] - Implement the many-to-many relationship between Person and Book using the intermediate class Rental, with has-many and belongs-to sides for Book, Rental, Person, and Rental.

### CODE REVIEW

**Comment by muneebulrehman** [comment]
Hi @ab-noori , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has no linter setup.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by ab-noori** [comment]
**Hi @muneebulrehman 👋,**
Thank you so much 🙏  for making me aware of the issue. I really forgot that.
Now I have checked and made it fix 🎯.

File: src/book.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class Book
+  attr_accessor :title, :author, :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+end
```
Comment (nahnahdev) [inline src/book.rb]: - [x] Great job so far @ab-noori  You have done an impressive job so far. However, per the project requirements, you needed to:

> Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following should be implemented:
> Create the has-many side of Book and Rental (a book has many rentals)

- Currently, this isn't being implemented in the `Book class`. I will suggest you include a method for adding rentals in the class. The book itself should be sent as a parameter to create the new rental thereby achieving the `has-many` association. :wink: 
- Here is a [link](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship%20def%20add_rental(person,%20date)) to an example of this implementation in a previous lesson :+1: 
- Example implementation

```ruby
  def add_rental(person, date)
    Rental.new(date, self, person)
  end
```

File: src/person.rb [hunk 2]
```diff
@@ -1,21 +1,25 @@
...
-  def initialize(name = "Unknown", age = 0, parent_permission = true)
+  def initialize(name: 'Unknown', age: 0, parent_permission: true)
+    super()
     @id = generate_id
-    @name = name
     @age = age
+    @name = name
     @parent_permission = parent_permission
+    @rentals = []
   end
```
Comment (nahnahdev) [inline src/person.rb]: - [x] Just as I mentioned in my previous comment, a similar implementation should be present in this `Person class` as well :+1: 
- [x]  Per the project requirements, you needed to:

> Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following should be implemented:
> Create the has-many side of Person and Rental (a person has many rentals).

- Currently, this isn't being implemented in the `Person class`. I will suggest you include a method for adding rentals in the class. The person itself should be sent as a parameter to create the new rental thereby achieving the `has-many` association. :wink: 
- Here is a [link](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship%20def%20add_rental(person,%20date)) to an example of this implementation in a previous lesson

**Review by nahnahdev — CHANGES_REQUESTED** [review]
## Project Status: CHANGES REQUIRED :recycle: 
Hi @ab-noori    , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights :fire: 
- Professional README :heavy_check_mark: 
- All required files present :heavy_check_mark: 
- Associations almost completed :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
- N/A

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ab-noori — COMMENTED** [review]
**Hi @NahnahAJ 👋,**
Thank you so much 🙏  for reviewing my project.
Your nice feedback is really appreciated 🙏.
I have checked and made them fix 🎯.

thank-you-gif-22

**Review by ridwanediallo — APPROVED** [review]
# STATUS: APPROVED 🟢
## Hi @ab-noori , 
Congratulations!GIF

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used GitHub flow correctly. ✔
- Well done implementing the changes requested.  🎉 🎉 🎉 ✔

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ab-noori — COMMENTED** [review]
**Hi @ridwanediallo 👋,**
Thank you so much 🙏  for reviewing and approving my project.
Your nice feedback is really appreciated 🙏.

Thank-You-Very-Much

**Pull request merged**